.github/workflows/ci.yml

```diff
@@ -49,8 +49,6 @@ jobs:
         os:
           - { label: ubuntu-latest, code: latest }
         compiler:
-          - { compiler: GNU10,  CC: gcc-10,   CXX: g++-10,     packages: gcc-10 g++-10 }
-          - { compiler: GNU11,  CC: gcc-11,   CXX: g++-11,     packages: gcc-11 g++-11 }
           - { compiler: GNU12,  CC: gcc-12,   CXX: g++-12,     packages: gcc-12 g++-12 }
           - { compiler: LLVM14, CC: clang-14, CXX: clang++-14, packages: clang-14 libomp-14-dev libclang-common-14-dev llvm-14-dev clang++-14 libc++-14-dev libc++1-14 libc++abi1-14 lld-14}
```

Comment:
Might as well bump to 15?

Reply:
Ideally you'd want to have a CI job for every supported compiler version (so at least GCC{12}, LLVM{12,13,14,15,16}, XCode{14.2,13.4,13.2}).
I don't really want to make a mess of an unrelated changes here,
i feel like adding coverage can be done separately.